src/ossia/protocols/midi/midi_protocol.cpp

```diff
@@ -531,20 +558,24 @@ std::vector<midi_info> midi_protocol::scan(libremidi::API api)
   std::vector<midi_info> vec;
 
   {
-    libremidi::midi_in in{api, ""};
+    libremidi::midi_in in {api, ""};
     auto portcount = in.get_port_count();
     for (auto i = 0u; i < portcount; i++)
     {
-      vec.emplace_back(midi_info::Type::Input, in.get_port_name(i), i);
+      vec.emplace_back(
+          midi_info::Type::Input, in.get_port_name(i), in.get_port_name(i), i,
```

Comment:
```
const std::string& name =  in.get_port_name(i); 
vec.emplace_back(midi_info::Type::Input, name, name, i, false);
```

this is because the get_port_name function is actually not just a getter but depending on the midi implementation can do some quite complex API calls, so better do them only once

Reply:
modified the code to avoid multiple call to `get_port_name`